core/src/com/mygdx/crazybomber/Map.java

```diff
@@ -0,0 +1,195 @@
+    private Sprite splash;
+    private SpriteBatch batch;
+    private Stage stage;
+    private Table table;
+    private TextButton buttonBack;
+    private Label heading;
+    private Skin skin;
+    private BitmapFont black, white;
+    private TextureAtlas atlas;
+    private Texture splashTexture;
+    private StateRepository stateRepository;
+   // private Sprite player;
+     private Player player;
+     private Texture playerTexture;
+    private int lmao = 0;
+    private int kek = 5;
+    boolean isPressedUP, isPressedDOWN, isPressedRIGHT, isPressedLEFT, isPressedW, isPressedS, isPressedD, isPressedA, isPressedSPACE, isPressedSHIFT;
+
+    public Map (Texture texture, final StateRepository stateRepository) {
+        this.splashTexture = texture; // gonna take in a texture instead of int
+        this.stateRepository = stateRepository;
+    }
+
+    @Override
+    public void show() {
+        playerTexture = new Texture("player.png");
+        batch = new SpriteBatch();
+        splash = new Sprite(splashTexture);
+        splash.setSize(Gdx.graphics.getWidth(), Gdx.graphics.getHeight());
+
+
+
+        //player
+        /*
+        Texture playerTexture = new Texture("player.png");
+        player = new Sprite(playerTexture);
+        player.setSize(50,50);
+        player.setPosition(615,345); */
+
+       // need to pass player the texture >> Texture playerTexture = new Texture("player.png");
+         player = new Player(0f, 0f, playerTexture);
+        player.setSize(50,50);
+        player.setPosition(615,600);
+
+
+        stage = new Stage();
+        Gdx.input.setInputProcessor(stage);
+        atlas = new TextureAtlas("ui/button.pack"); //defines the region of sprite image
+        skin = new Skin(atlas); //using this for the text button
+        table = new Table(skin); //table with the skin's atlas
+        //table.setBounds(500, 300, Gdx.graphics.getWidth(), Gdx.graphics.getHeight()); //sets the bounds to full screen
+        table.setBounds(1050, 600, 100,100 ); // coordinates of the table, size
+
+
+        white = new BitmapFont(Gdx.files.internal("font/white.fnt"), false); //this false is so the chars aren't flipped
+        black = new BitmapFont(Gdx.files.internal("font/black.fnt"), false);
+
+        TextButton.TextButtonStyle textButtonStyle = new TextButton.TextButtonStyle();
+        textButtonStyle.up = skin.getDrawable("button.up");
+        textButtonStyle.down = skin.getDrawable("button.down");
+        textButtonStyle.pressedOffsetX = 1; //when button is pressed, text moves to right by 1
+        textButtonStyle.pressedOffsetY = -1;
+        textButtonStyle.font = black;
+
+        buttonBack = new TextButton("BACK", textButtonStyle);
+        buttonBack.addListener(new ClickListener() {
+            @Override
+            public void clicked(InputEvent event, float x, float y) {
+                ((Game) Gdx.app.getApplicationListener()).setScreen(stateRepository.peek());
```

Comment:
should this be peek or pop?

Reply:
just peek

Reply:
can you explain why this is just peek when every other back button is pop() then peek()? why is the map not added to the state repository when the screen is set? because if we have a standard for buttonBack, we can have the code eventually in one place and we can use it instead of implementing it every time.

Reply:
because every state other than "Map" is pushed onto the stack and therefore requires them to be popped and then peeked to see the previous state. Map isn't pushed onto the state repository stack.